src/Google.Api.Gax/Platform.cs

```diff
@@ -112,10 +111,28 @@ public GcePlatformDetails(string metadataJson, string projectId, string instance
         public string InstanceId { get; }
 
         /// <summary>
-        /// The zone name where this GCE instance is running.
+        /// The zone where this GCE instance is running.
+        /// This is in the format <code>projects/&lt;project-number&gt;/zones/&lt;zone-name&gt;</code>.
         /// </summary>
         public string ZoneName { get; }
 
+        /// <summary>
+        /// The zone name where this GCE instance is running.
+        /// This is equivalent to the value of the <code>&lt;zone-name&gt;</code> part in
+        /// <see cref="ZoneName"/>
+        /// </summary>
+        public string Location
+        {
+            get
+            {
+                if (Platform.s_zoneTemplate.TryParseName(ZoneName, out var zoneResourceName))
+                {
+                    return zoneResourceName[1];
+                }
+                return null;
```

Comment:
`return ZoneName` instead of `null` here?

Reply:
I think I'll have to take a closer look at the existing code.

Reply:
I think null or throwing an exception is the right approach here. I would be tempted to throw `InvalidOperationException`, personally.

Reply:
(And we should have a test for it :)

Reply:
But then the only way a client has of not getting an Exception here is to pass the zone in the format projects/<project_number>/zones/<zone_name> in the constructor. And then, shouldn't we enforce that in the constructor instead of "here" (which would be breaking)?
After all, the Location property is being introduced in this PR, so we can define it: if we can extract the <zone_name> from ZoneName (oh the names) then Location = <zone_name> else Location = ZoneName. I think this is the least breaking change, and admittedly not the cleanest but it is not that bad either.
Anyway, happy to go the exception way if you still prefer it.

Reply:
(agreed with the test, I'll add them once we have decided)

Reply:
I think it's reasonable to define this property as only being available if the zone is in the right format. I don't think we should validate in the constructor - mostly to avoid it being a breaking change, as you say.

My second choice would be to return null to indicate that `ZoneName` is not in a format that we can extract the zone ID from... I don't mind that *much*, although I'd argue that an `InvalidOperationException` with a good message is more useful than a `NullReferenceException` for developers who don't read the docs with enough vigour.

@chrisdunelm thoughts?

Reply:
I think I prefer throwing; which should only occur if the user is constructing this class with metadata from a source that isn't GCE. Throwing with a good error message will better help the user to understand what's going on.

Reply:
Ok, will throw.